### PR TITLE
Support for codecov.io (OSIO#3559)

### PR DESCRIPTION
**Pull request by xcoulon**
Fixes openshiftio/openshift.io#3559

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @xcoulon for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-596-1`

**Comment by xcoulon** [comment]
@aslakknutsen @bartoszmajsak yes, I opened a PR to ignore the codecov.io config file: https://github.com/arquillian/ike-prow-plugins/compare/master...xcoulon:patch-2

**Comment by xcoulon** [comment]
[test]

File: cico_setup.sh [hunk 1]
```diff
@@ -96,7 +96,7 @@ function run_tests_with_coverage() {
 
   # Upload coverage to codecov.io
   cp tmp/coverage.mode* coverage.txt
-  bash <(curl -s https://codecov.io/bash) -X search -f coverage.txt -t ad12dad7-ebdc-47bc-a016-8c05fa7356bc #-X fix
+  bash <(curl -s https://codecov.io/bash) -X search -f coverage.txt -t fc7e9798-9ac2-4b06-af3e-c951fa579ed0 #-X fix
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply cico_setup.sh]: @aslakknutsen done in https://github.com/fabric8-services/fabric8-tenant/pull/596/commits/0f00dd87af230a40bfdc658c9aa13e88cb8df4ba

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-596-2`

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-596-3`

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-596-4`

**Pull request merged**